R/functions_io.R

```diff
@@ -464,3 +464,138 @@ ParsePlateInformation = function(cell_names, pattern='^(\\S+)_(\\d+)_([A-Z])(\\d
   
   return(plate_information)
 }
+
+#' Exports a Seurat object for visualisation with the cerebroApp (https://github.com/romanhaa/Cerebro).
+#' 
+#' @param sc A Seurat object.
+#' @param path File path for export.
+#' @param param Named list with parameters passed to the scrnaseq script.
+#' @param organism Species. Can be 'Hg', 'Mm', ... .
+#' @param assay Assay to export ('RNA').
+#' @param column_sample Metadata column containing the sample name ('orig.ident').
+#' @param column_cluster Metadata column containing the cluster identity ('seurat_clusters').
+#' @param column_ccphase Metadata column containing the cell cycle phase ('Phase').
+#' @param gene_lists Gene lists to include (NULL).
+#' @param marker_genes Marker genes table obtained from Seurat::FindMarkers (NULL).
+#' @param enriched_pathways List with enriched pathways results.
+#' @return TRUE if export was successful otherwise FALSE.
+ExportToCerebro = function(sc, path, param, project="scrnaseq", species, assay="RNA", column_sample="orig.ident", column_cluster="seurat_clusters", gene_lists = NULL, marker_genes=NULL, enriched_pathways=NULL, column_ccphase="Phase") {
+  
+  if (!(column_ccphase %in% colnames(sc[[]]))) column_ccphase = NULL
```

Comment:
Add a comment, something like: "Set to NULL if the requested CC phase is not part of the actual metadata."

Reply:
Done